packages/greedy_infomax/tests/unit/test_gradient_block.py

```diff
@@ -0,0 +1,62 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import unittest
+
+import torch
+import torch.nn as nn
+from torch.optim import Adam
+from torch.utils.data import DataLoader
+from torchvision.datasets import FakeData
+from torchvision.transforms import ToTensor
+
+from nupic.research.frameworks.greedy_infomax.models.utility_layers import GradientBlock
+
+
+class TestGradientBlock(unittest.TestCase):
+
+    def test_gradients_blocked(self):
```

Comment:
I recommend doing the following, which should't take more than 30 secs:

Add another test case that's identical to this one, except that you don't have a `GradientBlock` module in `model`, and test to see if the grads aren't None. That will be more convincing that your gradient block is working.

Reply:
Done